src/attr.rs

```diff
@@ -18,27 +21,39 @@ impl ToTokens for Display {
 }
 
 pub fn display(attrs: &[Attribute]) -> Result<Option<Display>> {
-    for attr in attrs {
-        if attr.path.is_ident("doc") {
-            let meta = attr.parse_meta()?;
-            let lit = match meta {
-                Meta::NameValue(syn::MetaNameValue {
-                    lit: syn::Lit::Str(lit),
-                    ..
-                }) => lit,
-                _ => unimplemented!(),
-            };
-
-            let lit = LitStr::new(lit.value().trim(), lit.span());
-
+    let attr = attrs
+        .iter()
+        .find(|attr| attr.path.is_ident("display"))
+        .or_else(|| attrs.iter().find(|attr| attr.path.is_ident("doc")));
+    if let Some(attr) = attr {
+        let meta = attr.parse_meta()?;
+        let lit = match meta {
+            Meta::NameValue(MetaNameValue { lit: Str(lit), .. }) => Some(lit),
+            Meta::NameValue(_) => unimplemented!("namevalue"),
```

Comment:
all of these unimplementeds should probably be removed and replaced with a catch all match arm that reports that the attribute was invalid with `compile_error!`

Reply:
I couldn't came up any good idea.

According to [the rust doc](https://doc.rust-lang.org/std/macro.compile_error.html) the ```compile_error!``` is supposed to use inside of macro. (I have never used this macro, so my interpretation might be wrong.)

If you just do

```rust
        let lit = match meta {
            Meta::NameValue(MetaNameValue { lit: Str(lit), .. }) => Some(lit),
            Meta::NameValue(_) => complie_error!("ERROR!!),
```

and simply compile everything, it gives a compile error. I thought writing a macro only for this match statement sounds overkill... (It's a good idea to show an error on compile time.)

I have also tried to return ```Ok``` and ```Err``` instead of ```Some``` and ```None```, but it might change current code a lot (for example, we can't use ```find_map``` if we decide to return Ok and Err).

Reply:
aah, so the way to return a compile_error! is to construct it with `quote!` and insert it in the generated code, rather than invoking it directly in the proc-macro. You might need to change the fn signature on this function to allow you to return a TokenStream.